dts/arm/silabs/siwg917.dtsi

```diff
@@ -59,6 +59,73 @@
 			current-speed = <115200>;
 			status = "disabled";
 		};
+
+		gpio: gpio@46130000 {
+			compatible = "silabs,siwx917-gpio";
+			reg = <0x46130000 0x1000>;
+			interrupts = <52 0>, <53 0>, <54 0>, <55 0>,
+				     <56 0>, <57 0>, <58 0>, <59 0>;
+			interrupt-names = "PIN0", "PIN1", "PIN2", "PIN3",
+					  "PIN4", "PIN5", "PIN6", "PIN7";
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			gpioa: gpio@0 {
+				compatible = "silabs,siwx917-gpio-port";
+				reg = <0>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			gpiob: gpio@1 {
+				compatible = "silabs,siwx917-gpio-port";
+				reg = <1>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			gpioc: gpio@2 {
+				compatible = "silabs,siwx917-gpio-port";
+				reg = <2>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			gpiod: gpio@3 {
+				compatible = "silabs,siwx917-gpio-port";
+				reg = <3>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			gpioulp: gpio@4 {
+				compatible = "silabs,siwx917-gpio-port";
+				reg = <4>;
+				interrupts = <18 0>;
+				interrupt-names = "ULP";
+
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			gpiouulp: gpio@5 {
+				compatible = "silabs,siwx917-gpio-uulp";
+				reg = <5>;
+				interrupts = <21 0>;
+				interrupt-names = "UULP";
+
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+		};
```

Comment:
I feel this structure is misleading. I believe `gpiouulp` do not share do not share anything with the parent. I feel `gpioulp` could also be a separated driver like `gpiouulp` (I am not a big fan of having a specific behavior if `interrupt-names == "ULP"`).
I also feel the `siwx917-gpio-port` are not independent devices (since they share their IRQs). Don't you think we should only declare `silabs,siwx917-gpio` with `#gpio-cells = <3>` (value `3` allows to pass 3 values for gpio instantiation: `gpios = <&gpio PORT_A 10 GPIO_ACTIVE_HIGH>`).

Reply:
Refactored.

Reply:
If my maths are correct, we now have exactly 32 pins in one `silabs,siwx917-gpio`.

Reply:
If I understand right, we now have exactly 32 GPIOs for one instance of `silabs,siwx917-gpio`

Reply:
`egpio0` (HP) covers pins from 0 to 57, split into 4 ports (and thus 4 struct device) of size 16,16,16,10.
`egpio1` (ULP) covers pins from 0 to 10.